### ISSUE TITLE
Create a top-level assembler entry point

### ISSUE THREAD

**Issue by bgyori** [sorgerlab/indra#105]
Currently the sequence mapping, grounding mapping, preassembly (duplicates, relations), mechanisms linking and belief engine all have to be imported and called one by one. 
It would be useful to define a top-level assembly API that would allow you to do all the internal steps of assembly in a single call with a set of options (policies) to define how the assembly should be done. Options would include things like: do_mechlinking, remove_sequence_error, grounding_agent_name_rewrite, etc. 
We need to think about the API carefully but it would be great to have this.

**Comment by johnbachman** [comment]
This reminds me a bit of the setup I created for model fitting, where I used a configuration file in YAML to select a dataset, identify the variables containing initial conditions, identify the model variations to try, the number of steps/walkers/temperatures to use for MCMC, etc. An individual fitting procedure for a single model was driven by one config file, but there were also higher-level config files specifying combinatorial sets of options that were used to create the low-level (single-fit config files).

I'm not sure if I would implement it exactly the same way again, but I agree that a programmatic setup for parameterizing the assembly process would be very useful, especially as we get into systematic validation/evaluation.
